src/transformers/tokenization_utils.py

```diff
@@ -176,70 +213,104 @@ def values(self):
 
         Args:
-            sentence: Index of the sentence relative to the batch provided to the tokenizer
-            char: Char index to get the relative token offsets
+            key (:obj:`int` or :obj:`slice`): Index of the token or tokens in the sequence
+            batch (:obj:`int`, , `optional`, defaults to :obj:`0`):
+                Index of the sequence in the batch (default to 0, the first sequence)
 
         Returns:
-            tuple: (token start, token end)
+            word_index (:obj:`int` or :obj:`List[int]`): index of the associated word or words in the input sequence.
 
         """
 
         if not self._encodings:
-            raise ValueError("char_to_token_offsets() is not available when using Python based tokenizers")
-        return self[sentence].char_to_token_offsets(char)
+            raise ValueError("token_to_word() is not available when using Python based tokenizers")
+        return self[batch].words[key]
 
-    def char_to_token(self, sentence: int, char: int) -> int:
-        """
-        Return the index of the token at position of the given char.
+    def word_to_token(self, key: Union[int, slice], batch: int = 0) -> Union[TokenSpan, List[TokenSpan]]:
+        """ Get the encoded tokens token(s) corresponding to word(s) in the sequence of the batch.
+
+            This method is particularly suited when the input sequences are provided as
+            pre-tokenized sequences (i.e. words are defined by the user). In this case it allows
+            to easily associate encoded tokens with provided tokenized words.
 
         Args:
-            sentence (int): Index of the sentence relative to the batch provided to the tokenizer
-            char (int): Char index to get the relative token offsets
+            key (:obj:`int` or :obj:`slice`): Index of the word or words in the sequence
+            batch (:obj:`int`, , `optional`, defaults to :obj:`0`):
+                Index of the sequence in the batch (default: 0)
 
         Returns:
-            int: Integer referring to the position of the token in the returned set of tokens for the sentence
+            token_span (:obj:`TokenSpan` or :obj:`List[TokenSpan]`):
+                Span of the associate token or tokens in the encoded sequence.
+                TokenSpan are NamedTuple with:
+                    start: index of the first token
+                    end: index of the token following the last token
         """
 
         if not self._encodings:
-            raise ValueError("char_to_token() is not available when using Python based tokenizers")
-        return self[sentence].char_to_token(char)
+            raise ValueError("word_to_token() is not available when using Python based tokenizers")
+        if isinstance(key, int):
+            return TokenSpan(*(self[batch].word_to_tokens(key)))
+        elif isinstance(key, slice):
+            return [TokenSpan(*(self[batch].word_to_tokens(i))) for i in key]
```

Comment:
slice is not iterable, I suspect this branch will throw an exception. 
slice usually just gives you something to do `range(slice.start, slice.end, slice.step)`.

Also, it seems to introduce an uncommon syntax here: `batch_encoding.word_to_token(slice(0, 4))`

Reply:
Ok, I'll just remove the slicing option for now and we can add it later if it seems useful.
I was not fully convinced by it either.